sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -345,6 +373,14 @@ ItemBuilder addIfNotDefault(
      * <p>Leaving the namespace unspecified will default to the registering instance's class.
      */
     ItemBuilder withNamespace(Class<?> namespace);
+
+    /**
+     * Adds an explicit namespace to the most-recently added display metadata. The namespace
+     * and key uniquely identify the display metadata.
+     *
+     * <p>Leaving the namespace unspecified will default to the registering instance's class.
+     */
+    ItemBuilder withNamespace(JavaClass namespace);
```

Comment:
My intent is to only apply to the most recently-added item; but I agree that it's ambiguous/misleading.

In C#, it's typical to use lambdas for tiered fluent-APIs:

    builder
        .add("foo1", "bar1", opt -> opt.withLabel("Foo 1"))
        .add("foo2", "bar2", opt -> opt
            .withNamespace("blarg")
            .withLinkUrl("http://goog"))
        .add("foo3", "bar3");

Another possibility is to let the user pass in an explicit 'options' object:

    builder
        .add("foo1", "bar1", DisplayData.Options.withLabel("Foo 1"))
        .add("foo2", "bar2", DisplayData.Options
            .withNamespace("blarg")
            .withLinkUrl("http://goog"))
        .add("foo3", "bar3");